torch/optim/sparse_adam.py

```diff
@@ -98,8 +98,8 @@ def make_sparse(values):
                 exp_avg_sq_update_values.add_(old_exp_avg_sq_values)
                 if amsgrad:
                     torch.max(old_max_exp_avg_sq_values, exp_avg_sq_update_values, out=old_max_exp_avg_sq_values)
+                    max_exp_avg_sq.put_(grad_indices, old_max_exp_avg_sq_values)
```

Comment:
I don't think that's correct either. `put_` will treat `grad_indices` as a flat list of linearized indices, which is not what you want in here. To be honest I would be fine with removing this feature from the sparse optimizer, at least to fix the issue quickly, and we can always add it back later.

Reply:
@apaszke removed for now 😞 will try to dig deeper into the API...